CRM/Core/BAO/Address.php

```diff
@@ -153,7 +153,7 @@ public static function add(&$params, $fixAddress) {
     }
 
     // (prevent chaining 1 and 3) CRM-21214
-    if (CRM_Utils_Array::value('master_id', $params)) {
+    if (!CRM_Utils_System::isNull($params['master_id'])) {
```

Comment:
Changing `CRM_Utils_Array::value` to `CRM_Utils_System::isNull` has another implication in that php will throw a warning if the array item is not set at all. To avoid this do

    if (isset($params['master_id']) && !CRM_Utils_System::isNull($params['master_id'])) {

Reply:
Cheers, updated.